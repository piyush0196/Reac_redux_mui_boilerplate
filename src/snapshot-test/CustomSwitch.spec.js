import TestRenderer from 'react-test-renderer'
import CustomSwitch from 'components/ui-kit/Switch'
import { theme } from 'theme/theme'
import { ThemeProvider } from '@mui/material'

describe('Component - CustomSwitch', () => {
  const wrappers = []
  const CustomSwitchJSX = ({
    onChange,
    label,
    size,
    labelPlacement,
    disabled
  }) => {
    return (
      <ThemeProvider theme={theme}>
        <CustomSwitch
          checked
          onChange={onChange}
          label={label}
          size={size}
          labelPlacement={labelPlacement}
          disabled={disabled}
        />
      </ThemeProvider>
    )
  }

  const labelPlacement = ['bottom', 'end', 'start', 'top']
  const sizes = ['small', 'medium', 'large']

  labelPlacement.forEach(placement => (
    wrappers.push(TestRenderer.create(
      <CustomSwitchJSX labelPlacement={placement} />
    ))
  ))
  sizes.forEach(size => (
    wrappers.push(TestRenderer.create(
      <CustomSwitchJSX size={size} />
    ))
  ))
  wrappers.push(TestRenderer.create(
    <CustomSwitchJSX onChange={() => {}} disabled />
  ))
  wrappers.push(TestRenderer.create(
    <CustomSwitchJSX onChange={() => {}} />
  ))
  wrappers.push(TestRenderer.create(
    <CustomSwitchJSX onChange={() => {}} />
  ))
  wrappers.push(TestRenderer.create(
    <CustomSwitchJSX onChange={() => {}} label='label' />
  ))
  wrappers.push(TestRenderer.create(
    <CustomSwitchJSX onChange={() => {}} />
  ))

  it('renders all checkboxes correctly', () => {
    wrappers.forEach(wrapper => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
