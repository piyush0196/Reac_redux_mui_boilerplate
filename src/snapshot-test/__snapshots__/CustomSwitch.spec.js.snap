// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - CustomSwitch renders all checkboxes correctly 1`] = `
<span
  className="MuiSwitch-root MuiSwitch-sizeMedium css-p2hj3q-MuiSwitch-root"
>
  <span
    className="MuiSwitch-switchBase MuiSwitch-colorSuccess Mui-checked MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorSuccess Mui-checked PrivateSwitchBase-root Mui-checked css-1edi9eb-MuiButtonBase-root-MuiSwitch-switchBase"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <input
      checked={true}
      className="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
      disabled={false}
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
    />
  </span>
  <span
    className="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
  />
</span>
`;

exports[`Component - CustomSwitch renders all checkboxes correctly 2`] = `
<span
  className="MuiSwitch-root MuiSwitch-sizeMedium css-p2hj3q-MuiSwitch-root"
>
  <span
    className="MuiSwitch-switchBase MuiSwitch-colorSuccess Mui-checked MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorSuccess Mui-checked PrivateSwitchBase-root Mui-checked css-1edi9eb-MuiButtonBase-root-MuiSwitch-switchBase"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <input
      checked={true}
      className="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
      disabled={false}
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
    />
  </span>
  <span
    className="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
  />
</span>
`;

exports[`Component - CustomSwitch renders all checkboxes correctly 3`] = `
<span
  className="MuiSwitch-root MuiSwitch-sizeMedium css-p2hj3q-MuiSwitch-root"
>
  <span
    className="MuiSwitch-switchBase MuiSwitch-colorSuccess Mui-checked MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorSuccess Mui-checked PrivateSwitchBase-root Mui-checked css-1edi9eb-MuiButtonBase-root-MuiSwitch-switchBase"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <input
      checked={true}
      className="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
      disabled={false}
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
    />
  </span>
  <span
    className="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
  />
</span>
`;

exports[`Component - CustomSwitch renders all checkboxes correctly 4`] = `
<span
  className="MuiSwitch-root MuiSwitch-sizeMedium css-p2hj3q-MuiSwitch-root"
>
  <span
    className="MuiSwitch-switchBase MuiSwitch-colorSuccess Mui-checked MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorSuccess Mui-checked PrivateSwitchBase-root Mui-checked css-1edi9eb-MuiButtonBase-root-MuiSwitch-switchBase"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <input
      checked={true}
      className="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
      disabled={false}
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
    />
  </span>
  <span
    className="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
  />
</span>
`;

exports[`Component - CustomSwitch renders all checkboxes correctly 5`] = `
<span
  className="MuiSwitch-root MuiSwitch-sizeSmall css-gta7z3-MuiSwitch-root"
>
  <span
    className="MuiSwitch-switchBase MuiSwitch-colorSuccess Mui-checked MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorSuccess Mui-checked PrivateSwitchBase-root Mui-checked css-1edi9eb-MuiButtonBase-root-MuiSwitch-switchBase"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <input
      checked={true}
      className="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
      disabled={false}
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
    />
  </span>
  <span
    className="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
  />
</span>
`;

exports[`Component - CustomSwitch renders all checkboxes correctly 6`] = `
<span
  className="MuiSwitch-root MuiSwitch-sizeMedium css-p2hj3q-MuiSwitch-root"
>
  <span
    className="MuiSwitch-switchBase MuiSwitch-colorSuccess Mui-checked MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorSuccess Mui-checked PrivateSwitchBase-root Mui-checked css-1edi9eb-MuiButtonBase-root-MuiSwitch-switchBase"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <input
      checked={true}
      className="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
      disabled={false}
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
    />
  </span>
  <span
    className="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
  />
</span>
`;

exports[`Component - CustomSwitch renders all checkboxes correctly 7`] = `
<span
  className="MuiSwitch-root MuiSwitch-sizeLarge css-21gmf3-MuiSwitch-root"
>
  <span
    className="MuiSwitch-switchBase MuiSwitch-colorSuccess Mui-checked MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorSuccess Mui-checked PrivateSwitchBase-root Mui-checked css-1edi9eb-MuiButtonBase-root-MuiSwitch-switchBase"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <input
      checked={true}
      className="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
      disabled={false}
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
    />
  </span>
  <span
    className="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
  />
</span>
`;

exports[`Component - CustomSwitch renders all checkboxes correctly 8`] = `
<span
  className="MuiSwitch-root MuiSwitch-sizeMedium css-p2hj3q-MuiSwitch-root"
>
  <span
    aria-disabled={true}
    className="MuiSwitch-switchBase MuiSwitch-colorSuccess Mui-checked Mui-disabled MuiButtonBase-root Mui-disabled MuiSwitch-switchBase MuiSwitch-colorSuccess Mui-checked Mui-disabled PrivateSwitchBase-root Mui-checked Mui-disabled css-1edi9eb-MuiButtonBase-root-MuiSwitch-switchBase"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={-1}
  >
    <input
      checked={true}
      className="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
      disabled={true}
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
    />
  </span>
  <span
    className="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
  />
</span>
`;

exports[`Component - CustomSwitch renders all checkboxes correctly 9`] = `
<span
  className="MuiSwitch-root MuiSwitch-sizeMedium css-p2hj3q-MuiSwitch-root"
>
  <span
    className="MuiSwitch-switchBase MuiSwitch-colorSuccess Mui-checked MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorSuccess Mui-checked PrivateSwitchBase-root Mui-checked css-1edi9eb-MuiButtonBase-root-MuiSwitch-switchBase"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <input
      checked={true}
      className="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
      disabled={false}
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
    />
  </span>
  <span
    className="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
  />
</span>
`;

exports[`Component - CustomSwitch renders all checkboxes correctly 10`] = `
<span
  className="MuiSwitch-root MuiSwitch-sizeMedium css-p2hj3q-MuiSwitch-root"
>
  <span
    className="MuiSwitch-switchBase MuiSwitch-colorSuccess Mui-checked MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorSuccess Mui-checked PrivateSwitchBase-root Mui-checked css-1edi9eb-MuiButtonBase-root-MuiSwitch-switchBase"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <input
      checked={true}
      className="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
      disabled={false}
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
    />
  </span>
  <span
    className="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
  />
</span>
`;

exports[`Component - CustomSwitch renders all checkboxes correctly 11`] = `
<span
  className="MuiSwitch-root MuiSwitch-sizeMedium css-p2hj3q-MuiSwitch-root"
>
  <span
    className="MuiSwitch-switchBase MuiSwitch-colorSuccess Mui-checked MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorSuccess Mui-checked PrivateSwitchBase-root Mui-checked css-1edi9eb-MuiButtonBase-root-MuiSwitch-switchBase"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <input
      checked={true}
      className="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
      disabled={false}
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
    />
  </span>
  <span
    className="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
  />
</span>
`;

exports[`Component - CustomSwitch renders all checkboxes correctly 12`] = `
<span
  className="MuiSwitch-root MuiSwitch-sizeMedium css-p2hj3q-MuiSwitch-root"
>
  <span
    className="MuiSwitch-switchBase MuiSwitch-colorSuccess Mui-checked MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorSuccess Mui-checked PrivateSwitchBase-root Mui-checked css-1edi9eb-MuiButtonBase-root-MuiSwitch-switchBase"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <input
      checked={true}
      className="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
      disabled={false}
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
    />
  </span>
  <span
    className="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
  />
</span>
`;
